plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AllFilesTable.java

```diff
@@ -0,0 +1,172 @@
+                .includeColumnStats();
+
+        tableScan.planFiles().forEach(fileScanTask -> {
+            DataFile dataFile = fileScanTask.file();
+
+            pagesBuilder.beginRow();
+            pagesBuilder.appendInteger(dataFile.content().id());
+            pagesBuilder.appendVarchar(dataFile.path().toString());
+            pagesBuilder.appendVarchar(dataFile.format().name());
+            pagesBuilder.appendBigint(dataFile.recordCount());
+            pagesBuilder.appendBigint(dataFile.fileSizeInBytes());
+            if (checkNonNull(dataFile.columnSizes(), pagesBuilder)) {
+                pagesBuilder.appendIntegerBigintMap(dataFile.columnSizes());
+            }
+            if (checkNonNull(dataFile.valueCounts(), pagesBuilder)) {
+                pagesBuilder.appendIntegerBigintMap(dataFile.valueCounts());
+            }
+            if (checkNonNull(dataFile.nullValueCounts(), pagesBuilder)) {
+                pagesBuilder.appendIntegerBigintMap(dataFile.nullValueCounts());
+            }
+            if (checkNonNull(dataFile.nanValueCounts(), pagesBuilder)) {
+                pagesBuilder.appendIntegerBigintMap(dataFile.nanValueCounts());
+            }
+            if (checkNonNull(dataFile.lowerBounds(), pagesBuilder)) {
+                pagesBuilder.appendIntegerVarcharMap(dataFile.lowerBounds().entrySet().stream()
+                        .collect(toImmutableMap(
+                                Map.Entry<Integer, ByteBuffer>::getKey,
+                                entry -> Transforms.identity(idToTypeMapping.get(entry.getKey())).toHumanString(
+                                        Conversions.fromByteBuffer(idToTypeMapping.get(entry.getKey()), entry.getValue())))));
+            }
+            if (checkNonNull(dataFile.upperBounds(), pagesBuilder)) {
+                pagesBuilder.appendIntegerVarcharMap(dataFile.upperBounds().entrySet().stream()
+                        .collect(toImmutableMap(
+                                Map.Entry<Integer, ByteBuffer>::getKey,
+                                entry -> Transforms.identity(idToTypeMapping.get(entry.getKey())).toHumanString(
+                                        Conversions.fromByteBuffer(idToTypeMapping.get(entry.getKey()), entry.getValue())))));
+            }
+            if (checkNonNull(dataFile.keyMetadata(), pagesBuilder)) {
+                pagesBuilder.appendVarbinary(Slices.wrappedBuffer(dataFile.keyMetadata()));
+            }
+            if (checkNonNull(dataFile.splitOffsets(), pagesBuilder)) {
+                pagesBuilder.appendBigintArray(dataFile.splitOffsets());
+            }
+            if (checkNonNull(dataFile.equalityFieldIds(), pagesBuilder)) {
+                pagesBuilder.appendIntegerArray(dataFile.equalityFieldIds());
+            }
+            pagesBuilder.endRow();
+        });
+
+        return pagesBuilder.build();
+    }
+
+    protected static boolean checkNonNull(Object object, PageListBuilder pagesBuilder) {
```

Comment:
private

Reply:
@alexjo2144 static can also be removed, as `buildPages` is non-static